app/code/Magento/Catalog/Model/Product/Url.php

```diff
@@ -19,54 +25,63 @@ class Url extends \Magento\Framework\DataObject
     /**
      * URL instance
      *
-     * @var \Magento\Framework\UrlFactory
+     * @var UrlFactory
      */
     protected $urlFactory;
 
     /**
-     * @var \Magento\Framework\Filter\FilterManager
+     * @var FilterManager
      */
     protected $filter;
 
     /**
      * Store manager
      *
-     * @var \Magento\Store\Model\StoreManagerInterface
+     * @var StoreManagerInterface
      */
     protected $storeManager;
 
     /**
-     * @var \Magento\Framework\Session\SidResolverInterface
+     * @var SidResolverInterface
      */
     protected $sidResolver;
 
     /**
-     * @var \Magento\UrlRewrite\Model\UrlFinderInterface
+     * @var UrlFinderInterface
      */
     protected $urlFinder;
 
     /**
-     * @param \Magento\Framework\UrlFactory $urlFactory
-     * @param \Magento\Store\Model\StoreManagerInterface $storeManager
-     * @param \Magento\Framework\Filter\FilterManager $filter
-     * @param \Magento\Framework\Session\SidResolverInterface $sidResolver
-     * @param UrlFinderInterface $urlFinder
-     * @param array $data
+     * @var ScopeConfigInterface
+     */
+    private $scopeConfig;
+
+    /**
+     * @param UrlFactory            $urlFactory
+     * @param StoreManagerInterface $storeManager
+     * @param FilterManager         $filter
+     * @param SidResolverInterface  $sidResolver
+     * @param UrlFinderInterface    $urlFinder
+     * @param array                 $data
+     * @param ScopeConfigInterface  $scopeConfig
      */
     public function __construct(
-        \Magento\Framework\UrlFactory $urlFactory,
-        \Magento\Store\Model\StoreManagerInterface $storeManager,
-        \Magento\Framework\Filter\FilterManager $filter,
-        \Magento\Framework\Session\SidResolverInterface $sidResolver,
+        UrlFactory $urlFactory,
+        StoreManagerInterface $storeManager,
+        FilterManager $filter,
+        SidResolverInterface $sidResolver,
         UrlFinderInterface $urlFinder,
-        array $data = []
+        array $data = [],
+        ScopeConfigInterface $scopeConfig = null
```

Comment:
For this code we are getting the following when running our static tests.

```
/var/www/html/app/code/Magento/Catalog/Model/Product/Url.php:23	The class Url has a coupling between objects value of 13. Consider to reduce the number of dependencies under 13.
```

Can you think of a way of refactoring these dependencies and class so that it is not such a high number?

Reply:
yep, I can add a different class as dependency instead of the ScopeConfigInterface and move the logic of getting the "product url use category" flag value to that class... I think that might reduce the number of coupled classes... will have a look as soon as I have time